core/txpool/legacypool/legacypool.go

```diff
@@ -202,7 +203,7 @@ type LegacyPool struct {
 	config      Config
 	chainconfig *params.ChainConfig
 	chain       BlockChain
-	gasTip      atomic.Pointer[big.Int]
+	gasTip      *uint256.Int
```

Comment:
We should retain the atomic pointer.  The gas tip is accessed without holding the pool mutex when transactions are added to the pool.

Reply:
modified, please review again